File: daten-scripten/schulen-im-wetteraukreis.py
import pandas as pd

def main():
    file_path = "../daten-generator/daten/Schulen.xlsx"
    df = pd.read_excel(file_path)

    wk_oeffSchulen = df[
        (df["di_LandkreisKz"] == 644000000000) & (df["di_Rechtsstellung"] == "ÖFF")
    ]

    grundschulen_typ = "G"
    weiterfuehrende_typen = ['HR', 'MSS', 'KGS', 'HRF', 'IGS', 'LER', 'GYM', 'GOS']
    berufsschulen_typ = "BS"

    gesamt = len(wk_oeffSchulen)
    grundschulen = len(wk_oeffSchulen[wk_oeffSchulen["di_Schultyp"] == grundschulen_typ])
    weiterführende = len(wk_oeffSchulen[wk_oeffSchulen["di_Schultyp"].isin(weiterfuehrende_typen)])
    berufsschulen = len(wk_oeffSchulen[wk_oeffSchulen["di_Schultyp"] == berufsschulen_typ])

    print(f"Gesamt: {gesamt}")
    print(f"Grundschulen: {grundschulen}")
    print(f"Weiterführende Schulen: {weiterführende}")
    print(f"Berufsschulen: {berufsschulen}")

if __name__ == "__main__":
    main()

File: r-daten-scripten/schulen-im-wetteraukreis.R
# ------------------------------------------------------------------------------
# Skript zur Zählung der öffentlichen Schulen im Wetteraukreis
# Unterscheidung nach Grundschulen, weiterführenden Schulen und Berufsschulen
# Verwendet: Schulen.xlsx
# ------------------------------------------------------------------------------
library(readxl)    # Excel einlesen
library(dplyr)     # Datenmanipulation
library(writexl)   # Excel schreiben

# Konstanten
source("common/konstanten.R")

# Schuldaten laden
schulen_df <- read_excel(schulen_path)

# Öffentliche Schulen im Wetteraukreis filtern
wk_oeffSchulen <- schulen_df %>%
  filter(
    di_LandkreisKz == wk_code,
    di_Rechtsstellung == "ÖFF"
  )

# Definition der Schultypen
grundschulen_typ <- "G"
weiterfuehrende_typen <- c("HR", "MSS", "KGS", "HRF", "IGS", "LER", "GYM", "GOS")
berufsschulen_typ <- "BS"

# Anzahl berechnen
gesamt <- nrow(wk_oeffSchulen)
grundschulen <- sum(wk_oeffSchulen$di_Schultyp == grundschulen_typ, na.rm = TRUE)
weiterfuehrende <- sum(wk_oeffSchulen$di_Schultyp %in% weiterfuehrende_typen, na.rm = TRUE)
berufsschulen <- sum(wk_oeffSchulen$di_Schultyp == berufsschulen_typ, na.rm = TRUE)

# Ergebnisse ausgeben
cat("Gesamt:", gesamt, "\n")
cat("Grundschulen:", grundschulen, "\n")
cat("Weiterführende Schulen:", weiterfuehrende, "\n")
cat("Berufsschulen:", berufsschulen, "\n")

# Ergebnis als Tabelle
result <- data.frame(
  Kategorie = c("Gesamt", "Grundschulen", "Weiterführende Schulen", "Berufsschulen"),
  Anzahl = c(gesamt, grundschulen, weiterfuehrende, berufsschulen)
)

# In Excel schreiben
if (!dir.exists("result")) dir.create("result")

write_xlsx(result, path = "result/schulen_anzahl.xlsx")

